src/librustdoc/html/static/main.js

```diff
@@ -1576,14 +1576,18 @@ function defocusSearchBar() {
         }
 
         function showResults(results) {
-            if (results.others.length === 1 &&
-                getCurrentValue("rustdoc-go-to-only-result") === "true") {
+            var search = getSearchElement();
+            if (results.others.length === 1
+                && getCurrentValue("rustdoc-go-to-only-result") === "true"
+                && (!search.firstChild || search.firstChild.innerText !== getSearchLoadingText()))
```

Comment:
In what cases will `!search.firstChild` happen?

Reply:
By default it's empty. So it's not empty when you already performed a search and "cancelled" it (meaning, you pressed ESC or remove all the content from the search input) or if there is the "loading search result" text when you arrive on the page with a search parameter in your URL.

Reply:
Does not look that straightforward to me but I understand after you explain.

Reply:
I'll add a comment then.